server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/IdentitySetUpdatePerformer.scala

```diff
@@ -0,0 +1,146 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.method
+
+import org.apache.james.jmap.api.identity.{IdentityNotFoundException, IdentityRepository, IdentityUpdateRequest}
+import org.apache.james.jmap.api.model.IdentityId
+import org.apache.james.jmap.core.Id.Id
+import org.apache.james.jmap.core.SetError.SetErrorDescription
+import org.apache.james.jmap.core.{Properties, SetError}
+import org.apache.james.jmap.json.IdentitySerializer
+import org.apache.james.jmap.mail.{IdentitySetRequest, InvalidPropertyException}
+import org.apache.james.jmap.method.IdentitySetUpdatePerformer.{IdentitySetUpdateResults, IdentityUpdate, IdentityUpdateParseException, UpdateFailure, UpdateResult, UpdateSuccess}
+import org.apache.james.mailbox.MailboxSession
+import play.api.libs.json.{JsObject, JsPath, JsValue, JsonValidationError}
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import java.util.UUID
+import javax.inject.Inject
+import scala.util.Try
+
+object IdentitySetUpdatePerformer {
+  sealed trait UpdateResult
+
+  case class UpdateSuccess(id: IdentityId) extends UpdateResult
+
+  case class UpdateFailure(id: UnparsedIdentitySetId, exception: Throwable) extends UpdateResult {
+    def asSetError: SetError = exception match {
+      case e: IdentityUpdateParseException => e.setError
+      case e: IdentityNotFoundException => SetError.notFound(SetErrorDescription(e.getMessage))
+      case e: InvalidPropertyException => SetError.invalidPatch(SetErrorDescription(s"${e.cause}"))
```

Comment:
```suggestion
      case e: InvalidPropertyException => SetError.invalidPatch(SetErrorDescription(e.cause.getMessage))
```

The string interpolation is clearly not needed here.